android/src/main/java/com/oblador/keychain/cipherStorage/CipherStorageBase.java

```diff
@@ -553,9 +553,9 @@ void initialize(@NonNull final Cipher cipher, @NonNull final Key key, @NonNull f
       throws GeneralSecurityException, IOException;
   }
 
-  /** Handler for configuring cipher by initialization data from input stream. */
-  public interface DecryptBytesHandler {
-    void initialize(@NonNull final Cipher cipher, @NonNull final Key key, @NonNull final InputStream input)
+  /** Handler for configuring cipher by initialization data from input. */
+  public interface DecryptBytesHandler<T> {
```

Comment:
I'm wondering if making this interface generic causes some confusion, especially when the different `CipherStorage` implementations don't explicitly state which type of `DecryptBytesHandler` they expect. Maybe would be better to not touch these common files too much, and just omit using the `handler` in the overridden `CipherStorageKeystoreAesCbc.decryptBytes`.

Reply:
Sure, I understand the concern. I considered just adding comments to specify the generics handled but didn't want to pigeonhole the interface too much. Also opted to not stray from the original implementation too much and therefore kept the `handler` but your suggestion makes sense also and I'm happy to make that change.